### DISCUSSION TITLE
Why GPL not AGPL?

### DISCUSSION THREAD

**Discussion by zeidspex** [dani-garcia/vaultwarden#1681] [General]
I have noticed that the license for this project is GPL, while -imo- AGPL is a more appropriate license since the primary use of this code is remote access through a server.

Is there any particular reason why GPL is used? Maybe the authors can change the license now, since the longer the project waits the harder it will be to change the license.

**Comment by dani-garcia** [comment]
Well, mostly because I didn't put much thought into it initially as I didn't expect the project to become popular in the first place. If I was starting the project today I'd probably choose the AGPL too, so I'm not personally against changing it, but to change the license now we'd have to get every contributor to agree, which would be a pretty big task

**Reply by zeidspex** [reply]
Yeah, I know that the more you wait the harder it becomes. There are 71 contributors as far as I can tell. I think you can simply create an issue or send an email to all the contributors to ask for permission to change the license. I think most (and probably all) the contributors would agree. I think it is worth the effort to try and make the change happen.

Also, I am not a lawyer, but I think you don't even need to get permission from all contributors. Only those who made 'substantial' contributions (maybe 50 lines of code or more). Anything less than that would be considered fair use (in my non-expert opinion).

_Edit:_ I checked. You have (at most) 28 contributors who have contributed 50 lines of code or more. That's quite a manageable number. And I also read around a little bit. There is good legal reasons why minor contributors do not block license changes, so you don't need to ask all your contributors to agree.

**Comment by pepa65** [comment]
I don't see a particular advantage for using the AGPL in this case. Why do you think it is better, @zeidspex ?

**Reply by zeidspex** [reply]
Since this project is designed mainly to be hosted and served over the network, the GPL terms will not trigger. The code at this point may as well be licensed as BSD or MIT. To ensure that copyleft terms are relevant, we need AGPL. Otherwise, individuals and\or companies can host a Bitwarden-compatible service, modify the code, and not contribute any changes back.

Remember, GPL is only triggered on distribution. And serving over the network is NOT considered distribution; Which is why AGPL was created.

**Comment by pepa65** [comment]
Thanks for educating my ignorance, I was confusing it with the LGPL.
In that case, it would be really good to change the license!

**Comment by BlackDex** [comment]
I think we could at least try to contact all contributors. Also all previous versions will still be GPLv3, and i think al major contributors will probably agree with the change.

Maybe it is also an idea to enforce some `code of conduct` to the project?

**Reply by zeidspex** [reply]
I think a code of conduct is orthogonal to the issue of the license. It can be added later if or when needed. The license is a more urgent issue, as the longer we wait the harder it gets.

While previous versions will be GPLv3, there is still a lot more work and features that need to be added, so those could benefit from a new license.

Anyhow, yes, I agree. Contacting contributors is the way to go. Based on some reading, I would suggest adding an issue and a corresponding file to the project titled "ConsentToAGPL.md". And have contributors add their names through pull requests to this file in order to document their consent. The names of the contributors required to provide consent can be listed and tagged in an issue, and when all contributors have consented, then it is a done deal. In the meanwhile, there can be a notice in the README file that any new contributions must be accompanied with consent to AGPL.
